### PR TITLE
Encrypt zdb password

### LINKED ISSUE #396: zdb provision: password for zdb namespace should be encrypted in the reseravtion
At the moment the password for a 0-DB namespace reservation is in clear text into the reservation.
This needs to change. The password can be encrypted with the public key of the node.

### PR DESCRIPTION
**Pull request by muhamadazmy**
Fixes #396

### CODE REVIEW

**Review by zaibon — APPROVED** [review]
LGTM. only question is why did you choose to use base58 to encode the password after encryption ?
In the network reservation we do use hex to encode the password, see: https://github.com/threefoldtech/zos/blob/master/tools/tfuser/cmds_network.go#L150

I think we should be consistent in the way we encrypt/encode things

**Comment by muhamadazmy** [comment]
@zaibon node-id uses base58 to encrypt the node-id from a public key, I just used the same encoding.

File: tools/tfuser/main.go [hunk 1]
```diff
@@ -253,6 +253,10 @@ func main() {
 									Name:  "password, p",
 									Usage: "optional password",
 								},
+								cli.StringFlag{
```
Comment (zaibon) [inline tools/tfuser/main.go:256]: I don't think we should add this flag here.
The node ID is already specify in the provision command.
So I think the encryption needs to happen during provision command. 
- load json
- if it's a zdb reservation, load the zdb data 
- encrypt the password using the node ID
- sign the full reservation with encrypted password
- send the signed reservation to BCDB

**Review by zaibon — CHANGES_REQUESTED** [review]

File: tools/tfuser/main.go (hunk 1 above)
Comment (muhamadazmy) [reply tools/tfuser/main.go:256]: Yeah, I considered that first but then i realized
- The password will be unencrypted in the schema file
- The provision need to know the type of the reservation and have special handling for the zdb case.

**Pull request merged**